src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Document.cs

```diff
@@ -107,5 +112,42 @@ private void ScanTagInDocumentContext()
                 tagBlock.Dispose();
             }
         }
+
+        private bool CurrentTagBlockExpectsJavascriptContent()
```

Comment:
Method name is incorrect. Returns a value indicating the current `<script>` tag does not expect HTML. But it could expect an infinite number of other content types, not just JavaScript. Suggest reversing the various conditions where this is used, return `false` by default, and name the method `CurrentScriptTagExpectsHtml()`.